plugins/catalog-backend/src/next/DefaultLocationStore.ts

```diff
@@ -111,6 +111,39 @@ export class DefaultLocationStore implements LocationStore, EntityProvider {
     });
   }
 
+  async refreshLocation(type: string): Promise<string> {
+    if (!this.connection) {
+      throw new Error('location store is not initialized');
+    }
+
+    const location = await this.db<DbLocationsRow>('locations')
+      .where({ type })
+      .select('id');
+
+    if (!location[0]) {
+      throw new NotFoundError(`Found no location with type ${type}`);
+    }
+
+    const id = location[0].id;
+
+    const isRefreshRowExists = await this.db<DbRefreshStateRow>('refresh_state')
+      .where('entity_id', id)
+      .select();
+
+    let result = `Location found, refresh row not found for type ${type}, please wait a while and try again`;
```

Comment:
Please advise on messaging here- as I understand it, a row will be inserted eventually/soon if the type is new and doesn't have one yet, but I'm not confident in that assessment.